// @bobaboard/ao3.js/dist/cjs/types/entities.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkWarnings = exports.WorkWarningStatus = exports.WorkCategory = exports.WorkRatings = void 0;
var WorkRatings;
(function (WorkRatings) {
    WorkRatings["NOT_RATED"] = "Not Rated";
    WorkRatings["GENERAL_AUDIENCES"] = "General Audiences";
    WorkRatings["TEEN_AND_UP_AUDIENCES"] = "Teen And Up Audiences";
    WorkRatings["MATURE"] = "Mature";
    WorkRatings["EXPLICIT"] = "Explicit";
})(WorkRatings = exports.WorkRatings || (exports.WorkRatings = {}));
var WorkCategory;
(function (WorkCategory) {
    WorkCategory["FF"] = "F/F";
    WorkCategory["FM"] = "F/M";
    WorkCategory["GEN"] = "Gen";
    WorkCategory["MM"] = "M/M";
    WorkCategory["MULTI"] = "Multi";
    WorkCategory["OTHER"] = "Other";
})(WorkCategory = exports.WorkCategory || (exports.WorkCategory = {}));
var WorkWarningStatus;
(function (WorkWarningStatus) {
    WorkWarningStatus["NO_WARNINGS_APPLY"] = "Author indicated no warnings apply";
    WorkWarningStatus["CHOOSE_NOT_TO_WARN"] = "Author chose not to warn";
    WorkWarningStatus["EXTERNAL"] = "External work";
    WorkWarningStatus["HAS_WARNING"] = "Work has one or more warning";
})(WorkWarningStatus = exports.WorkWarningStatus || (exports.WorkWarningStatus = {}));
var WorkWarnings;
(function (WorkWarnings) {
    WorkWarnings["GRAPHIC_VIOLENCE"] = "Graphic depictions of violence";
    WorkWarnings["MAJOR_CHARACTER_DEATH"] = "Major character death";
    WorkWarnings["NONCON"] = "Rape/non-con";
    WorkWarnings["UNDERAGE"] = "Underage";
})(WorkWarnings = exports.WorkWarnings || (exports.WorkWarnings = {}));
