// @bobaboard/ao3.js/dist/esm/utils/user.js
import cheerio from "cheerio";
import axios from "axios";
export const getProfileLink = (userName) => `https://archiveofourown.org/users/${encodeURI(userName)}/profile`;
export const getProfile = async (userName) => {
    return cheerio.load((await axios.get(getProfileLink(userName))).data);
};
export const getProfileName = ($userProfile) => {
    return $userProfile(".user.profile .header h2").text().trim();
};
export const getProfilePseuds = ($userProfile) => {
    const pseuds = $userProfile("dd.pseuds").text().concat(", ");
    return pseuds.slice(0, -2);
};
// slice here prevents pseuds from getting an extra ", " at the end
export const getProfileJoined = ($userProfile) => {
    const dds = $userProfile(".meta dd:not(.email, .location, .birthday, .pseuds)").text();
    return dds.slice(0, 10);
};
//slice here cuts the date off before it would run into the user ID
export const getProfileID = ($userProfile) => {
    const dds = $userProfile(".meta dd:not(.email):not(dt.location+dd):not(.birthday):not(.pseuds)").text();
    return dds.slice(10);
};
export const getProfileBio = ($userProfile) => {
    return $userProfile(".bio .userstuff").html();
};
export const getProfileEmail = ($userProfile) => {
    return $userProfile("dd.email").text();
};
export const getProfileLocation = ($userProfile) => {
    return $userProfile("dt.location+dd").text();
};
export const getProfileBday = ($userProfile) => {
    return $userProfile("dd.birthday").text();
};
//TODO: Pull information (Works/Series/Bookmarks/Collections/Gifts) from navigation actions maybe?
