// @bobaboard/ao3.js/dist/cjs/utils/tags.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getCanonical = exports.getTagName = exports.isCanonical = exports.isCommon = exports.hasMergers = exports.getTagCategory = exports.getTagPage = exports.getTagUrl = void 0;
const cheerio_1 = __importDefault(require("cheerio"));
const axios_1 = __importDefault(require("axios"));
const getTagUrl = (tagName) => `https://archiveofourown.org/tags/${encodeURI(tagName)
    .replace("/", "*s*")
    .replace("&", "*a*")
    .replace(".", "*d*")}`;
exports.getTagUrl = getTagUrl;
const getTagPage = async (tagName) => {
    return cheerio_1.default.load((await axios_1.default.get((0, exports.getTagUrl)(tagName))).data);
};
exports.getTagPage = getTagPage;
const getTagCategory = ($tagPage) => {
    // This will look similar to "This tag belongs to the Character Category."
    const categorySentence = $tagPage($tagPage(".tag.profile > p")[0]).text();
    const category = categorySentence.match(/This tag belongs to the (.+) Category/)[1];
    return category.toLowerCase();
};
exports.getTagCategory = getTagCategory;
const hasMergers = ($tagPage) => {
    return $tagPage(".merger").length > 0;
};
exports.hasMergers = hasMergers;
const isCommon = ($tagPage) => {
    const categorySentence = $tagPage($tagPage(".tag.profile > p")[0]).text();
    // TODO: check whether my assumption that all tags that have mergers have a parent that's
    // been marked as common.
    return (categorySentence.includes("It's a common tag.") || (0, exports.hasMergers)($tagPage));
};
exports.isCommon = isCommon;
const isCanonical = ($tagPage) => {
    return (0, exports.isCommon)($tagPage) && !(0, exports.hasMergers)($tagPage);
};
exports.isCanonical = isCanonical;
const getTagName = ($tagPage) => {
    return $tagPage($tagPage(".tag.profile h2")[0]).text();
};
exports.getTagName = getTagName;
const getCanonical = ($tagPage) => {
    if ((0, exports.isCanonical)($tagPage)) {
        return (0, exports.getTagName)($tagPage);
    }
    if (!(0, exports.hasMergers)($tagPage)) {
        return null;
    }
    return $tagPage($tagPage(".merger a.tag")).text();
};
exports.getCanonical = getCanonical;
