// @bobaboard/ao3.js/dist/cjs/utils/user.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getProfileBday = exports.getProfileLocation = exports.getProfileEmail = exports.getProfileBio = exports.getProfileID = exports.getProfileJoined = exports.getProfilePseuds = exports.getProfileName = exports.getProfile = exports.getProfileLink = void 0;
const cheerio_1 = __importDefault(require("cheerio"));
const axios_1 = __importDefault(require("axios"));
const getProfileLink = (userName) => `https://archiveofourown.org/users/${encodeURI(userName)}/profile`;
exports.getProfileLink = getProfileLink;
const getProfile = async (userName) => {
    return cheerio_1.default.load((await axios_1.default.get((0, exports.getProfileLink)(userName))).data);
};
exports.getProfile = getProfile;
const getProfileName = ($userProfile) => {
    return $userProfile(".user.profile .header h2").text().trim();
};
exports.getProfileName = getProfileName;
const getProfilePseuds = ($userProfile) => {
    const pseuds = $userProfile("dd.pseuds").text().concat(", ");
    return pseuds.slice(0, -2);
};
exports.getProfilePseuds = getProfilePseuds;
// slice here prevents pseuds from getting an extra ", " at the end
const getProfileJoined = ($userProfile) => {
    const dds = $userProfile(".meta dd:not(.email, .location, .birthday, .pseuds)").text();
    return dds.slice(0, 10);
};
exports.getProfileJoined = getProfileJoined;
//slice here cuts the date off before it would run into the user ID
const getProfileID = ($userProfile) => {
    const dds = $userProfile(".meta dd:not(.email):not(dt.location+dd):not(.birthday):not(.pseuds)").text();
    return dds.slice(10);
};
exports.getProfileID = getProfileID;
const getProfileBio = ($userProfile) => {
    return $userProfile(".bio .userstuff").html();
};
exports.getProfileBio = getProfileBio;
const getProfileEmail = ($userProfile) => {
    return $userProfile("dd.email").text();
};
exports.getProfileEmail = getProfileEmail;
const getProfileLocation = ($userProfile) => {
    return $userProfile("dt.location+dd").text();
};
exports.getProfileLocation = getProfileLocation;
const getProfileBday = ($userProfile) => {
    return $userProfile("dd.birthday").text();
};
exports.getProfileBday = getProfileBday;
//TODO: Pull information (Works/Series/Bookmarks/Collections/Gifts) from navigation actions maybe?
